### ISSUE TITLE
Connection handshake failed with CloudAMQP

### ISSUE THREAD

**Issue by mattborghi** [JuliaComputing/AMQPClient.jl#37]
Hi, I am trying to connect to an instance of [`CloudAMQP`](https://www.cloudamqp.com/) but I am getting  the following error

<details><summary>ERROR: LoadError: AMQPClient.AMQPClientException("Connection handshake failed")</summary>
<p>

```sh
Stacktrace:
 [1] connection(; virtualhost::String, host::String, port::Int64, framemax::Int64, heartbeat::Int64, send_queue_size::Int64, auth_params::Dict{String,Any}, channelmax::Int64, connect_timeout::Int64, amqps::Nothing) at /home/borghi/.julia/packages/AMQPClient/P6CBt/src/protocol.jl:574
 [2] connect(; cfg::Dict{String,String}) at /home/borghi/Desktop/umc/task-queues-microservice/results/Worker/src/Worker.jl:56
 [3] connect() at /home/borghi/Desktop/umc/task-queues-microservice/results/Worker/src/Worker.jl:43
 [4] top-level scope at /home/borghi/Desktop/umc/task-queues-microservice/results/run_worker.jl:9
 [5] include(::Function, ::Module, ::String) at ./Base.jl:380
 [6] include(::Module, ::String) at ./Base.jl:368
 [7] exec_options(::Base.JLOptions) at ./client.jl:296
 [8] _start() at ./client.jl:506
```

</p>
</details>

I have no issues connecting to a `localhost` server and in the cloud I tested it using `pika` package from `python` successfully. 

I leave a snippet of the testing code (it's nothing special really)

```julia
auth_params = Dict{String,Any}("MECHANISM" => "AMQPLAIN", "LOGIN" => login, "PASSWORD" => password)
conn = connection(; virtualhost=vhost, host, port, auth_params, amqps=nothing)
```

where `login` and `password` are the string credentials. `vhost` equals `login`. `host` is the string url of the form `xxxx.rmq.cloudamqp.com` and `port=5672`.

> As a sidenote it would be really nice if there would be a method like `pika` that let's you connect using a URL like [here](https://pika.readthedocs.io/en/stable/modules/parameters.html#urlparameters).

Thanks in advance, Matias.

**Comment by mattborghi** [comment]
As a temporary fix (I hope), I ended up using [`PyCall.jl`](https://github.com/JuliaPy/PyCall.jl) and [`pika`](https://pypi.org/project/pika/) for all the `AMPQ` calls and works well for my use case, but it would be awesome if someone knows how to resolve the issue with the Julia client I explained above.

**Comment by tanmaykm** [comment]
Thanks for reporting. This should be fixed by #40.
I shall tag a new version shortly.
